diff --git a/arch/arm/boot/dts/nxp/imx/Makefile b/arch/arm/boot/dts/nxp/imx/Makefile
index 12e091dc4c4c..e0c92dc8cd80 100644
--- a/arch/arm/boot/dts/nxp/imx/Makefile
+++ b/arch/arm/boot/dts/nxp/imx/Makefile
@@ -297,7 +297,10 @@ dtb-$(CONFIG_SOC_IMX6Q) += \
        imx6qp-yapp4-crux-plus.dtb \
        imx6qp-yapp4-pegasus-plus.dtb \
        imx6qp-zii-rdu2.dtb \
-       imx6s-dhcom-drc02.dtb
+       imx6s-dhcom-drc02.dtb \
+       mxxf1.dtb \
+       mxxf1-hw4.dtb \
+       mxxf1-tsup.dtb
 dtb-$(CONFIG_SOC_IMX6SL) += \
        imx6sl-evk.dtb \
        imx6sl-evk-ldo.dtb \