diff --git a/arch/arm/boot/dts/nxp/imx/Makefile b/arch/arm/boot/dts/nxp/imx/Makefile
index 12e091dc4c4c..e0c92dc8cd80 100644
--- a/arch/arm/boot/dts/nxp/imx/Makefile
+++ b/arch/arm/boot/dts/nxp/imx/Makefile
@@ -297,7 +297,10 @@ dtb-$(CONFIG_SOC_IMX6Q) += \
 	imx6qp-yapp4-crux-plus.dtb \
 	imx6qp-yapp4-pegasus-plus.dtb \
 	imx6qp-zii-rdu2.dtb \
-	imx6s-dhcom-drc02.dtb
+	imx6s-dhcom-drc02.dtb \
+	mxxf1.dtb \
+	mxxf1-hw4.dtb \
+	mxxf1-tsup.dtb
 dtb-$(CONFIG_SOC_IMX6SL) += \
 	imx6sl-evk.dtb \
 	imx6sl-evk-ldo.dtb \
diff --git a/arch/arm/boot/dts/nxp/imx/mxxf1-common.dtsi b/arch/arm/boot/dts/nxp/imx/mxxf1-common.dtsi
new file mode 100644
index 000000000000..007806501cd7
--- /dev/null
+++ b/arch/arm/boot/dts/nxp/imx/mxxf1-common.dtsi
@@ -0,0 +1,850 @@
+/*
+ * Copyright DATA RESPONS AS
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	aliases {
+		mxcfb0 = &mxcfb1;
+		mmc0 = &usdhc3;
+		mmc1 = &usdhc4;
+		rtc0 = &main_rtc;
+	};
+
+	memory {
+		reg = <0x10000000 0x40000000>;
+	};
+
+	chosen {
+		stdout-path = &uart1;
+	};
+
+	backlight  {
+		compatible = "pwm-backlight";
+		pwms = <&pwm1 0 500000>;
+		brightness-levels = < 0 1 2 3 4 5 6 8  11 16  23  32 48
+							  64 81 100 121 144 169 196 225
+							  256 289 324 361 400 441 484 529
+							  576 625 676 729 784 841 900 961
+							  1024 1089 1156 1225 1296 1369 1444 1521
+							  1600 1681 1764 1849 1936 2025 2116 2209 >;
+		default-brightness-level = <32>;
+		enable-gpios = <&gpio1 2 GPIO_ACTIVE_HIGH>;
+		power-supply = <&reg_5v>;
+		status = "okay";
+	};
+
+	mxcfb1: fb@0 {
+		compatible = "fsl,mxc_sdc_fb";
+		disp_dev = "ldb";
+		interface_pix_fmt = "RGB24";
+		default_bpp = <32>;
+		int_clk = <0>;
+		late_init = <0>;
+		status = "okay";
+	};
+
+	regulators {
+		compatible = "simple-bus";
+
+		reg_usb_otg_vbus: usb_otg_vbus {
+			compatible = "regulator-fixed";
+			regulator-name = "usb_otg_vbus";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			gpio = <&gpio3 22 0>;
+			enable-active-high;
+		};
+
+		reg_usb_h1_vbus: usb_h1_vbus {
+			compatible = "regulator-fixed";
+			regulator-name = "usb_h1_vbus";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			gpio = <&gpio1 29 0>;
+			enable-active-high;
+		};
+		reg_usb_p0_vbus: usb_p0_vbus {
+			compatible = "regulator-fixed";
+			regulator-name = "usb_p0_vbus";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			gpio = <&gpio3 28 0>;
+			enable-active-high;
+			regulator-boot-on;
+			regulator-always-on;
+		};
+
+		reg_usb_p1_vbus: usb_p1_vbus {
+			compatible = "regulator-fixed";
+			regulator-name = "usb_p1_vbus";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			gpio = <&gpio2 3 0>;
+			enable-active-high;
+			regulator-boot-on;
+			regulator-always-on;
+		};
+
+		reg_usb_p2_vbus: usb_p2_vbus {
+			compatible = "regulator-fixed";
+			regulator-name = "usb_p2_vbus";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			gpio = <&gpio2 7 0>;
+			enable-active-high;
+			regulator-boot-on;
+			regulator-always-on;
+		};
+
+		reg_usb_p3_vbus: usb_p3_vbus {
+			compatible = "regulator-fixed";
+			regulator-name = "usb_p3_vbus";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			gpio = <&gpio6 9 0>;
+			enable-active-high;
+			regulator-boot-on;
+			regulator-always-on;
+		};
+
+		reg_5v: ext_5v {
+			compatible = "regulator-fixed";
+			regulator-name = "5V";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			gpio = <&gpio6 10 0>;
+			enable-active-high;
+			regulator-boot-on;
+			regulator-always-on;
+		};
+
+		reg_touch: touch_pwr {
+			compatible = "regulator-fixed";
+			regulator-name = "TOUCH";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+			gpio = <&gpio1 4 0>;
+			enable-active-high;
+			regulator-always-on;
+		};
+
+
+		reg_lcd: lcd_pwr {
+			compatible = "regulator-fixed";
+			regulator-name = "LCD";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+			gpio = <&gpio6 15 0>;
+			enable-active-high;
+			regulator-always-on;
+		};
+
+	};
+
+	pwm-buzzer {
+		compatible = "pwm-buzzer";
+		pwms = <&pwm3 0 434782>;
+		status = "okay";
+	};
+
+	gpio-keys {
+		compatible = "gpio-keys";
+		power {
+			label = "Power Button";
+			gpios = <&gpio3 29 1>;
+			linux,code = <KEY_POWER>; /* KEY_POWER */
+			gpio-key,wakeup;
+		};
+
+		dimm-up {
+			label = "Brightness Up";
+			gpios = <&gpio4 7 1>;
+			linux,code = <KEY_BRIGHTNESSUP>; /* KEY_BRIGHTNESSUP */
+		};
+
+		dimm-down {
+			label = "Brightness Down";
+			gpios = <&gpio4 6 1>;
+			linux,code = <KEY_BRIGHTNESSDOWN>; /* KEY_BRIGHTNESSDOWN */
+		};
+	};
+
+/*
+	user-gpios {
+		gpio-lan2-eewp {
+			gpios = <&gpio6 14 GPIO_ACTIVE_LOW>;
+			status = "okay";
+			value = <0>;
+		};
+
+		gpio-eeprom-wp {
+			gpios = <&gpio1 5 GPIO_ACTIVE_LOW>;
+			status = "okay";
+			value = <0>;
+		};
+
+		gpio-spinor-wp {
+			gpios = <&gpio6 11 GPIO_ACTIVE_LOW>;
+			status = "okay";
+			value = <1>;
+		};
+
+		gpio-fan {
+			gpios = <&gpio1 18 GPIO_ACTIVE_HIGH>;
+			status = "okay";
+			value = <0>;
+		};
+
+		gpio-ext-buzz {
+			gpios = <&gpio1 19 GPIO_ACTIVE_HIGH>;
+			status = "okay";
+			value = <0>;
+		};
+
+		gpio-phy-rst {
+			gpios = <&gpio1 25 GPIO_ACTIVE_LOW>;
+			status = "okay";
+			value = <1>;
+		};
+		gpio-hw0 {
+			gpios = <&gpio1 15 GPIO_ACTIVE_HIGH>;
+			status = "okay";
+		};
+
+		gpio-hw1 {
+			gpios = <&gpio1 14 GPIO_ACTIVE_HIGH>;
+			status = "okay";
+		};
+
+		gpio-hw2 {
+			gpios = <&gpio1 13 GPIO_ACTIVE_HIGH>;
+			status = "okay";
+		};
+
+		gpio-hw3 {
+			gpios = <&gpio1 12 GPIO_ACTIVE_HIGH>;
+			status = "okay";
+		};
+
+		gpio-lan2-off {
+		gpios = <&gpio3 23 GPIO_ACTIVE_LOW>;
+		value = <1>;
+		status = "okay";
+		};
+
+	};
+*/
+};
+
+
+&gpio1 {
+	gpio-line-names = "", "", "", "", "", "eeprom-wp", "", "",
+		"", "", "", "", "hwrev3", "hwrev2", "hwrev1", "hwrev0",
+		"", "", "fan", "buzzer";
+};
+
+&gpio3 {
+	lan2_off {
+		gpio-hog;
+		gpios = <23 0>;
+		output-high;
+	};
+};
+
+&gpio6 {
+	gpio-line-names = "", "", "", "", "", "", "", "",
+		"", "", "", "spiflash-wp", "", "", "lan2-wp";
+	lan2_wp {
+		gpio-hog;
+		gpios = <14 0>;
+		output-high;
+	};
+};
+
+&ecspi1 {
+	fsl,spi-num-chipselects = <1>;
+	cs-gpios = <&gpio3 19 GPIO_ACTIVE_LOW>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_ecspi1>;
+	status = "okay";
+	nodma;
+
+	norflash: m25p80@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "jedec,spi-nor";
+		spi-max-frequency = <20000000>;
+		reg = <0>;
+		use-large-blocks;
+	};
+};
+
+&norflash {
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		bootloader: partition@0 {
+			label = "bootloader";
+			reg = <0 0x00100000>;
+		};
+		boot_env: partition@00100000 {
+			label = "boot-env";
+			reg = <0x00100000 0x40000>;
+		};
+		user: partition@00140000 {
+			label = "factory";
+			reg = <0x00140000 0x280000>;
+		};
+		state_storage: partition@003C0000 {
+			label = "barebox-state";
+			reg = <0x003C0000 0x40000>;
+		};
+	};
+};
+
+&gpc {
+	fsl,ldo-bypass = <0>; /* use ldo-bypass, u-boot will check it and configure */
+	fsl,wdog-reset = <2>; /* watchdog select of reset source */
+};
+
+&wdog1 {
+	status = "okay";
+};
+
+&wdog2 {
+	status = "disabled";
+	fsl,ext-reset-output;
+};
+
+&i2c2 {
+	clock-frequency = <100000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c2>;
+	status = "okay";
+
+		pmic: pfuze100@8 {
+		compatible = "fsl,pfuze100";
+		reg = <0x08>;
+
+		regulators {
+			sw1a_reg: sw1ab {
+				regulator-min-microvolt = <300000>;
+				regulator-max-microvolt = <1875000>;
+				regulator-boot-on;
+				regulator-always-on;
+				regulator-ramp-delay = <6250>;
+			};
+
+			sw1c_reg: sw1c {
+				regulator-min-microvolt = <300000>;
+				regulator-max-microvolt = <1875000>;
+				regulator-boot-on;
+				regulator-always-on;
+				regulator-ramp-delay = <6250>;
+			};
+
+			sw2_reg: sw2 {
+				regulator-min-microvolt = <800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			sw3a_reg: sw3a {
+				regulator-min-microvolt = <400000>;
+				regulator-max-microvolt = <1975000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			sw3b_reg: sw3b {
+				regulator-min-microvolt = <400000>;
+				regulator-max-microvolt = <1975000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			sw4_reg: sw4 {
+				regulator-min-microvolt = <800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			swbst_reg: swbst {
+				regulator-min-microvolt = <5000000>;
+				regulator-max-microvolt = <5150000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			snvs_reg: vsnvs {
+				regulator-min-microvolt = <1000000>;
+				regulator-max-microvolt = <3000000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			vref_reg: vrefddr {
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			vgen1_reg: vgen1 {
+				regulator-min-microvolt = <800000>;
+				regulator-max-microvolt = <1550000>;
+			};
+
+			vgen2_reg: vgen2 {
+				regulator-min-microvolt = <800000>;
+				regulator-max-microvolt = <1550000>;
+			};
+
+			vgen3_reg: vgen3 {
+				regulator-min-microvolt = <2500000>;
+				regulator-max-microvolt = <2500000>;
+			};
+
+			vgen4_reg: vgen4 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+
+			vgen5_reg: vgen5 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+
+			vgen6_reg: vgen6 {
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+		};
+	};
+
+	eeprom@50 {
+		compatible = "atmel,24c16";
+		reg = <0x50>;
+		pagesize = <16>;
+	};
+
+};
+
+&i2c1 {
+	status = "disabled";
+};
+
+&iomuxc {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_hog_1>;
+
+	hog {
+		pinctrl_hog_1: hoggrp-1 {
+			fsl,pins = <
+				MX6QDL_PAD_GPIO_4__GPIO1_IO04   0x80000000
+				MX6QDL_PAD_GPIO_5__GPIO1_IO05   0x80000000
+				MX6QDL_PAD_SD1_CMD__GPIO1_IO18  0x80000000
+				MX6QDL_PAD_SD1_DAT2__GPIO1_IO19 0x80000000
+				MX6QDL_PAD_NANDF_D0__GPIO2_IO00 0x80000000
+				MX6QDL_PAD_NANDF_D1__GPIO2_IO01 0x80000000
+				MX6QDL_PAD_NANDF_D2__GPIO2_IO02 0x80000000
+				MX6QDL_PAD_NANDF_D3__GPIO2_IO03 0x80000000
+				MX6QDL_PAD_NANDF_D7__GPIO2_IO07 0x80000000
+				MX6QDL_PAD_NANDF_ALE__GPIO6_IO08 0x80000000
+				MX6QDL_PAD_NANDF_CLE__GPIO6_IO07 0x80000000
+				MX6QDL_PAD_GPIO_0__CCM_CLKO1    0x130b0
+				MX6QDL_PAD_NANDF_CS2__GPIO6_IO15 0x80000000
+				MX6QDL_PAD_NANDF_CS3__GPIO6_IO16 0x80000000
+				MX6QDL_PAD_EIM_D26__GPIO3_IO26  0x80000000
+				MX6QDL_PAD_EIM_CS1__GPIO2_IO24  0x80000000
+				MX6QDL_PAD_ENET_RXD0__GPIO1_IO27 0x80000000
+				MX6QDL_PAD_EIM_A25__GPIO5_IO02  0x80000000
+				MX6QDL_PAD_ENET_TXD1__GPIO1_IO29 0x80000000
+				MX6QDL_PAD_EIM_D28__GPIO3_IO28  0x80000000
+				MX6QDL_PAD_NANDF_CS0__GPIO6_IO11 0x80000000
+				MX6QDL_PAD_NANDF_CS1__GPIO6_IO14 0x80000000
+				MX6QDL_PAD_EIM_EB3__GPIO2_IO31  0x80000000
+				MX6QDL_PAD_EIM_DA9__GPIO3_IO09  0x80000000
+				MX6QDL_PAD_GPIO_2__GPIO1_IO02   0x80000000
+				MX6QDL_PAD_SD3_RST__GPIO7_IO08  0x80000000
+				MX6QDL_PAD_GPIO_1__WDOG2_B      0x1b000
+				MX6QDL_PAD_GPIO_2__GPIO1_IO02   0x80000000
+				MX6QDL_PAD_NANDF_RB0__GPIO6_IO10 0x80000000
+				MX6QDL_PAD_NANDF_WP_B__GPIO6_IO09 0x80000000
+				MX6QDL_PAD_CSI0_DATA_EN__GPIO5_IO20 0x80000000
+				MX6QDL_PAD_EIM_D30__USB_H1_OC 0x80000000
+				MX6QDL_PAD_SD2_DAT0__GPIO1_IO15 0x80000000
+				MX6QDL_PAD_SD2_DAT1__GPIO1_IO14 0x80000000
+				MX6QDL_PAD_SD2_DAT2__GPIO1_IO13 0x80000000
+				MX6QDL_PAD_SD2_DAT3__GPIO1_IO12 0x80000000
+				MX6QDL_PAD_EIM_D29__GPIO3_IO29  0x80000000
+				MX6QDL_PAD_KEY_COL0__GPIO4_IO06 0x80000000
+				MX6QDL_PAD_KEY_ROW0__GPIO4_IO07 0x80000000
+				MX6QDL_PAD_KEY_ROW1__GPIO4_IO09 0x80000000
+			>;
+		};
+	};
+
+	pcie {
+		pinctrl_pcie: pciegrp {
+			fsl,pins = <
+				MX6QDL_PAD_GPIO_17__GPIO7_IO12	0x1b0b0
+				MX6QDL_PAD_EIM_D23__GPIO3_IO23  0x1b0b0
+			>;
+		};
+	};
+
+	ecspi1 {
+		pinctrl_ecspi1: ecspi1grp {
+			fsl,pins = <
+				MX6QDL_PAD_EIM_D17__ECSPI1_MISO	0x100b1
+				MX6QDL_PAD_EIM_D18__ECSPI1_MOSI	0x100b1
+				MX6QDL_PAD_EIM_D16__ECSPI1_SCLK	0x100b1
+				MX6QDL_PAD_EIM_D19__GPIO3_IO19	0x100b1
+			>;
+		};
+	};
+
+	ecspi2 {
+		pinctrl_ecspi2: ecspi2grp {
+			fsl,pins = <
+				MX6QDL_PAD_EIM_OE__ECSPI2_MISO	0x100b1
+				MX6QDL_PAD_EIM_CS1__ECSPI2_MOSI	0x100b1
+				MX6QDL_PAD_EIM_CS0__ECSPI2_SCLK	0x100b1
+				MX6QDL_PAD_DISP0_DAT18__GPIO5_IO12	0x100b1
+			>;
+		};
+	};
+
+	uart1 {
+		pinctrl_uart1: uart1grp {
+			fsl,pins = <
+				MX6QDL_PAD_CSI0_DAT10__UART1_TX_DATA	0x1b0b1
+				MX6QDL_PAD_CSI0_DAT11__UART1_RX_DATA	0x1b0b1
+			>;
+		};
+	};
+
+	uart2 {
+		pinctrl_uart2: uart2grp {
+			fsl,pins = <
+				MX6QDL_PAD_GPIO_7__UART2_TX_DATA	0x1b0b1
+				MX6QDL_PAD_GPIO_8__UART2_RX_DATA	0x1b0b1
+			>;
+		};
+	};
+
+	uart3 {
+		pinctrl_uart3: uart3grp {
+			fsl,pins = <
+				MX6QDL_PAD_EIM_D24__UART3_TX_DATA	0x1b0b1
+				MX6QDL_PAD_EIM_D25__UART3_RX_DATA	0x1b0b1
+			>;
+		};
+	};
+
+	uart4 {
+		pinctrl_uart4: uart4grp {
+			fsl,pins = <
+				MX6QDL_PAD_KEY_COL0__UART4_TX_DATA	0x1b0b1
+				MX6QDL_PAD_KEY_ROW0__UART4_RX_DATA	0x1b0b1
+			>;
+		};
+	};
+
+	pwm1 {
+		pinctrl_pwm1: pwm1grp {
+			fsl,pins = <
+				MX6QDL_PAD_SD1_DAT3__PWM1_OUT		0x1b0b1
+			>;
+		};
+	};
+
+	pwm3 {
+		pinctrl_pwm3: pwm3grp {
+			fsl,pins = <
+				MX6QDL_PAD_SD1_DAT1__PWM3_OUT 0x1b0b1
+			>;
+		};
+	};
+
+	i2c2 {
+		pinctrl_i2c2: i2c2grp {
+			fsl,pins = <
+				MX6QDL_PAD_KEY_COL3__I2C2_SCL       0x4001b8b1
+				MX6QDL_PAD_KEY_ROW3__I2C2_SDA       0x4001b8b1
+			 >;
+		};
+	};
+
+	i2c3 {
+		pinctrl_i2c3: i2c3grp {
+			fsl,pins = <
+				MX6QDL_PAD_GPIO_3__I2C3_SCL 0x4001b8b1
+				MX6QDL_PAD_GPIO_6__I2C3_SDA 0x4001b8b1
+			>;
+		};
+	};
+
+	usbotg {
+		pinctrl_usbotg_lm: usbotggrp-lm {
+			fsl,pins = <
+				MX6QDL_PAD_ENET_RX_ER__USB_OTG_ID 	0x17059
+				MX6QDL_PAD_EIM_D21__USB_OTG_OC		0x17059
+				MX6QDL_PAD_EIM_D22__GPIO3_IO22		0x80000000
+
+			>;
+		};
+	};
+
+	usdhc3 {
+		pinctrl_usdhc3: usdhc3grp {
+			fsl,pins = <
+				MX6QDL_PAD_SD3_CMD__SD3_CMD		0x17059
+				MX6QDL_PAD_SD3_CLK__SD3_CLK		0x10059
+				MX6QDL_PAD_SD3_DAT0__SD3_DATA0		0x17059
+				MX6QDL_PAD_SD3_DAT1__SD3_DATA1		0x17059
+				MX6QDL_PAD_SD3_DAT2__SD3_DATA2		0x17059
+				MX6QDL_PAD_SD3_DAT3__SD3_DATA3		0x17059
+
+			>;
+		};
+	};
+
+	usdhc4 {
+		pinctrl_usdhc4: usdhc4grp {
+			fsl,pins = <
+				MX6QDL_PAD_SD4_CMD__SD4_CMD		0x17059
+				MX6QDL_PAD_SD4_CLK__SD4_CLK		0x10059
+				MX6QDL_PAD_SD4_DAT0__SD4_DATA0		0x17059
+				MX6QDL_PAD_SD4_DAT1__SD4_DATA1		0x17059
+				MX6QDL_PAD_SD4_DAT2__SD4_DATA2		0x17059
+				MX6QDL_PAD_SD4_DAT3__SD4_DATA3		0x17059
+				MX6QDL_PAD_SD4_DAT4__SD4_DATA4      0x17059
+				MX6QDL_PAD_SD4_DAT5__SD4_DATA5      0x17059
+				MX6QDL_PAD_SD4_DAT6__SD4_DATA6      0x17059
+				MX6QDL_PAD_SD4_DAT7__SD4_DATA7      0x17059
+			>;
+		};
+	};
+
+	enet {
+		pinctrl_enet_lm: enetgrp-lm2 {
+			fsl,pins = <
+				MX6QDL_PAD_ENET_CRS_DV__GPIO1_IO25    0x1b0b0	/* Phy reset (low) */
+				MX6QDL_PAD_ENET_RXD1__GPIO1_IO26 	  0x1b0b0     	/* Phy IRQ (low) */
+				MX6QDL_PAD_ENET_MDIO__ENET_MDIO       0x1b0b0
+				MX6QDL_PAD_ENET_MDC__ENET_MDC         0x1b0b0
+				MX6QDL_PAD_RGMII_TXC__RGMII_TXC       0x1b0b0
+				MX6QDL_PAD_RGMII_TD0__RGMII_TD0       0x1b0b0
+				MX6QDL_PAD_RGMII_TD1__RGMII_TD1       0x1b0b0
+				MX6QDL_PAD_RGMII_TD2__RGMII_TD2       0x1b0b0
+				MX6QDL_PAD_RGMII_TD3__RGMII_TD3       0x1b0b0
+				MX6QDL_PAD_RGMII_TX_CTL__RGMII_TX_CTL 0x1b0b0
+				MX6QDL_PAD_ENET_REF_CLK__ENET_TX_CLK  0x4001b0a8
+				MX6QDL_PAD_RGMII_RXC__RGMII_RXC       0x1b0b0
+				MX6QDL_PAD_RGMII_RD0__RGMII_RD0       0x1b0b0
+				MX6QDL_PAD_RGMII_RD1__RGMII_RD1       0x1b0b0
+				MX6QDL_PAD_RGMII_RD2__RGMII_RD2       0x1b0b0
+				MX6QDL_PAD_RGMII_RD3__RGMII_RD3       0x1b0b0
+				MX6QDL_PAD_RGMII_RX_CTL__RGMII_RX_CTL 0x1b0b0
+				MX6QDL_PAD_ENET_TX_EN__GPIO1_IO28     0x1b0b0
+			>;
+		};
+	};
+
+	can1 {
+		pinctrl_flexcan1: flexcan1grp {
+			fsl,pins = <
+				MX6QDL_PAD_GPIO_8__FLEXCAN1_RX    0x80000000
+				MX6QDL_PAD_GPIO_7__FLEXCAN1_TX    0x80000000
+			>;
+		};
+	};
+
+	can2 {
+		pinctrl_flexcan2: flexcan2grp {
+			fsl,pins = <
+				MX6QDL_PAD_KEY_COL4__FLEXCAN2_TX    0x80000000
+				MX6QDL_PAD_KEY_ROW4__FLEXCAN2_RX    0x80000000
+			>;
+		};
+	};
+};
+
+&ldb {
+	status = "okay";
+
+	lvds-channel@0 {
+		fsl,data-mapping = "spwg";
+		fsl,data-width = <24>;
+		status = "okay";
+		primary;
+		crtc = "ipu1-di0";
+
+		display-timings {
+			native-mode = <&timing0>;
+			timing0: hsd100pxn1 {
+				clock-frequency = <65000000>;
+				hactive = <1024>;
+				vactive = <768>;
+				hback-porch = <220>;
+				hfront-porch = <40>;
+				vback-porch = <21>;
+				vfront-porch = <7>;
+				hsync-len = <60>;
+				vsync-len = <10>;
+			};
+		};
+	};
+
+	lvds-channel@1 {
+		fsl,data-mapping = "spwg";
+		fsl,data-width = <24>;
+		status = "disabled";
+		crtc = "ipu2-di1";
+		display-timings {
+			native-mode = <&timing1>;
+			timing1: hsd100pxn1 {
+				clock-frequency = <65000000>;
+				hactive = <1024>;
+				vactive = <768>;
+				hback-porch = <220>;
+				hfront-porch = <40>;
+				vback-porch = <21>;
+				vfront-porch = <7>;
+				hsync-len = <60>;
+				vsync-len = <10>;
+			};
+		};
+	};
+};
+
+
+
+&pcie {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pcie>;
+	reset-gpio = <&gpio7 12 GPIO_ACTIVE_LOW>;
+	status = "okay";
+	vph-supply = <&vgen3_reg>;
+};
+
+&uart1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart1>;
+	status = "okay";
+};
+
+&usbh1 {
+	vbus-supply = <&reg_usb_h1_vbus>;
+	status = "okay";
+	over-current-active-low;
+};
+
+&usbotg {
+	vbus-supply = <&reg_usb_otg_vbus>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usbotg_lm>;
+	over-current-active-low;
+	status = "okay";
+};
+
+&usdhc2 {
+	status = "disabled";
+};
+
+&usdhc3 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usdhc3>;
+	cd-gpios = <&gpio2 0 GPIO_ACTIVE_LOW>;
+	wp-gpios = <&gpio2 1 GPIO_ACTIVE_HIGH>;
+	bus-width = <4>;
+	no-1-8-v;
+	keep-power-in-suspend;
+	enable-sdio-wakeup;
+	status = "okay";
+};
+
+&usdhc4 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usdhc4>;
+	bus-width = <8>;
+	non-removable;
+	no-1-8-v;
+	keep-power-in-suspend;
+	status = "okay";
+};
+
+&can1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_flexcan1>;
+	status = "okay";
+};
+
+&can2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_flexcan2>;
+	status = "okay";
+};
+
+&sata {
+	status = "okay";
+};
+
+&pwm3 {
+	#pwm-cells = <2>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pwm3>;
+	status = "okay";
+};
+
+&clks {
+	fsl,ldb-di0-parent = <&clks IMX6QDL_CLK_PLL2_PFD0_352M>;
+	fsl,ldb-di1-parent = <&clks IMX6QDL_CLK_PLL2_PFD0_352M>;
+};
+
+&snvs_poweroff {
+	status = "okay";
+};
+
+&usbphy1 {
+	tx-d-cal = <0x5>;
+};
+
+&usbphy2 {
+	tx-d-cal = <0x5>;
+};
+
+&pwm1 {
+	#pwm-cells = <2>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_pwm1>;
+	status = "okay";
+};
+
+&ocotp {
+	status = "okay";
+};
+
+&tempmon {
+	status = "okay";
+};
+
+&src {
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/nxp/imx/mxxf1-hw4.dts b/arch/arm/boot/dts/nxp/imx/mxxf1-hw4.dts
new file mode 100644
index 000000000000..c37e707313a4
--- /dev/null
+++ b/arch/arm/boot/dts/nxp/imx/mxxf1-hw4.dts
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2015 DATA RESPONS AS
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+/dts-v1/;
+
+#include "imx6q.dtsi"
+#include "mxxf1-common.dtsi"
+
+/ {
+    model = "DR imx6q Maritime Computer - Midlife Upgrade";
+    compatible = "fsl,imx6q";
+};
+
+&i2c3 {
+    clock-frequency = <100000>;
+    pinctrl-names = "default";
+    pinctrl-0 = <&pinctrl_i2c3>;
+    status = "okay";
+
+    main_rtc: rtc@68 {
+        compatible = "stm,m41t82";
+        reg = <0x68>;
+        };
+
+    adc7417@28 {
+        compatible = "adi,ad7417";
+        reg = <0x28>;
+        };
+};
+
+&fec {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_enet_lm>;
+	phy-mode = "rgmii";
+	status = "okay";
+	phy-reset-duration = <10>;
+	phy-reset-gpios = <&gpio1 25 GPIO_ACTIVE_LOW>;
+	phy-handle = <&miphy>;
+	phy-supply = <&vgen3_reg>;
+    txen-skew-ps = <0>;
+	txc-skew-ps = <3000>;
+	rxdv-skew-ps = <0>;
+	rxc-skew-ps = <3000>;
+	rxd0-skew-ps = <0>;
+	rxd1-skew-ps = <0>;
+	rxd2-skew-ps = <0>;
+	rxd3-skew-ps = <0>;
+	txd0-skew-ps = <0>;
+	txd1-skew-ps = <0>;
+	txd2-skew-ps = <0>;
+	txd3-skew-ps = <0>;
+	mdio {
+	   #address-cells = <1>;
+       #size-cells = <0>;
+	   miphy: ethernet-phy@1 {
+	       compatible = "ethernet-phy-ieee802.3-c22";
+	       reg = <0x1>;
+	       interrupt-parent = <&gpio1>;
+	       interrupts = <28 IRQ_TYPE_LEVEL_LOW>;
+	   };
+	};
+};
diff --git a/arch/arm/boot/dts/nxp/imx/mxxf1-tsup.dts b/arch/arm/boot/dts/nxp/imx/mxxf1-tsup.dts
new file mode 100644
index 000000000000..aca9478c31fa
--- /dev/null
+++ b/arch/arm/boot/dts/nxp/imx/mxxf1-tsup.dts
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2015 DATA RESPONS AS
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+/dts-v1/;
+
+#include "imx6q.dtsi"
+#include "mxxf1-common.dtsi"
+
+/ {
+    model = "DR imx6q Maritime Computer - new touch";
+    compatible = "fsl,imx6q";
+
+};
+
+&i2c3 {
+    clock-frequency = <100000>;
+    pinctrl-names = "default";
+    pinctrl-0 = <&pinctrl_i2c3>;
+    status = "okay";
+
+    egalax_ts@2a {
+        compatible = "eeti,egalax_i2c";
+        reg = <0x2a>;
+        interrupt-parent = <&gpio6>;
+        interrupts = <8 IRQ_TYPE_LEVEL_LOW>;
+        int-gpios = <&gpio6 8 0>;
+        disable-suspend = <1>;
+        };
+
+    main_rtc: rtc@68 {
+        compatible = "stm,m41t82";
+        reg = <0x68>;
+        };
+
+    adc7417@28 {
+        compatible = "adi,ad7417";
+        reg = <0x28>;
+        };
+};
+
+&fec {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_enet_lm>;
+	phy-mode = "rgmii-id";
+	status = "okay";
+	phy-reset-duration = <10>;
+	phy-reset-gpios = <&gpio1 25 GPIO_ACTIVE_LOW>;
+	phy-handle = <&arphy>;
+	phy-supply = <&vgen6_reg>;
+	mdio {
+	   #address-cells = <1>;
+       #size-cells = <0>;
+	   arphy: ethernet-phy@1 {
+	       compatible = "ethernet-phy-ieee802.3-c22";
+	       reg = <0x1>;
+	       interrupt-parent = <&gpio1>;
+	       interrupts = <26 IRQ_TYPE_EDGE_FALLING>;
+	   };
+	};
+};
diff --git a/arch/arm/boot/dts/nxp/imx/mxxf1.dts b/arch/arm/boot/dts/nxp/imx/mxxf1.dts
new file mode 100644
index 000000000000..de1a28b1eae5
--- /dev/null
+++ b/arch/arm/boot/dts/nxp/imx/mxxf1.dts
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2015 DATA RESPONS AS
+ * Copyright 2012 Freescale Semiconductor, Inc.
+ * Copyright 2011 Linaro Ltd.
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+/dts-v1/;
+
+#include "imx6q.dtsi"
+#include "mxxf1-common.dtsi"
+#include <dt-bindings/net/qca-ar803x.h>
+
+/ {
+	model = "DR imx6q Maritime Computer";
+	compatible = "fsl,imx6q";
+};
+
+&i2c3 {
+	clock-frequency = <100000>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c3>;
+	status = "okay";
+
+	egalax_ts@4 {
+		compatible = "eeti,egalax_ts";
+		reg = <0x04>;
+		interrupt-parent = <&gpio6>;
+		interrupts = <8 2>;
+		wakeup-gpios = <&gpio6 8 0>;
+		disable-suspend = <1>;
+		};
+
+	main_rtc: rtc@68 {
+		compatible = "stm,m41t82";
+		reg = <0x68>;
+		};
+
+	adc7417@28 {
+		compatible = "adi,ad7417";
+		reg = <0x28>;
+		};
+};
+
+&fec {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_enet_lm>;
+	phy-mode = "rgmii-id";
+	status = "okay";
+	phy-reset-duration = <10>;
+	phy-reset-gpios = <&gpio1 25 GPIO_ACTIVE_LOW>;
+	phy-handle = <&arphy>;
+	phy-supply = <&vgen6_reg>;
+	mdio {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		arphy: ethernet-phy@1 {
+			compatible = "ethernet-phy-id004d.d074";
+		   	qca,clk-out-frequency = <125000000>;
+		   	qca,clk-out-strength = <AR803X_STRENGTH_FULL>;
+		   	qca,keep-pll-enabled;
+			at803x,eee-disabled;
+			reg = <0x1>;
+			interrupt-parent = <&gpio1>;
+			interrupts = <26 IRQ_TYPE_EDGE_FALLING>;
+	   };
+	};
+};
